### PR TITLE
1.x: fix SyncOnSubscribe not signalling onError if the generator crashes

### PR DESCRIPTION
**Pull request by akarnokd**
Discovered in a [stackoverflow](http://stackoverflow.com/questions/35001387/how-to-handle-error-in-generatestate-in-synconsubscribe-rxjava) question.

### CODE REVIEW

File: src/main/java/rx/observables/SyncOnSubscribe.java [hunk 1]
```diff
@@ -363,7 +373,12 @@ private boolean tryUnsubscribe() {
         }
 
         private void doUnsubscribe() {
-            parent.onUnsubscribe(state);
+            try {
+                parent.onUnsubscribe(state);
+            } catch (Throwable e) {
```
Comment (artem-zinnatullin) [inline src/main/java/rx/observables/SyncOnSubscribe.java]: Can you test that too?

Comment (akarnokd) [inline src/main/java/rx/observables/SyncOnSubscribe.java]: Not really, this exception has nowhere to go and hijacking the plugins is a nightmare.

Comment (artem-zinnatullin) [inline src/main/java/rx/observables/SyncOnSubscribe.java]: That's sad, would be nice to have a way to test such cases.

On Tue, 26 Jan 2016, 00:44 David Karnok <notifications@github.com> wrote:

> In src/main/java/rx/observables/SyncOnSubscribe.java
> <https://github.com/ReactiveX/RxJava/pull/3644#discussion_r50760936>:
>
> > @@ -363,7 +373,12 @@ private boolean tryUnsubscribe() {
> >          }
> >
> >          private void doUnsubscribe() {
> > -            parent.onUnsubscribe(state);
> > +            try {
> > +                parent.onUnsubscribe(state);
> > +            } catch (Throwable e) {
>
> Not really, this exception has nowhere to go and hijacking the plugins is
> a nightmare.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ReactiveX/RxJava/pull/3644/files#r50760936>.
>
-- 

@artem_zin

**Comment by artem-zinnatullin** [comment]
:+1:

**Comment by stealthcode** [comment]
:+1:

**Comment by akarnokd** [comment]
@stealthcode Do you have time to review #3631, see what else can be included into 1.1.1 and hopefully release it this week?

**Comment by stealthcode** [comment]
Sure I'll see what we can do.

**Pull request merged**